lib/publiccloud/instance.pm

```diff
@@ -42,10 +44,51 @@ sub run_ssh_command {
 
     $args{timeout} //= SSH_TIMEOUT;
 
-    my $ssh_cmd = sprintf('ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -oLogLevel=ERROR -i "%s" "%s@%s" -- %s',
-        $self->ssh_key, $self->username, $self->public_ip, $args{cmd});
+    my $ssh_cmd = sprintf('ssh %s -i "%s" "%s@%s" -- %s',
+        $self->ssh_opts, $self->ssh_key, $self->username, $self->public_ip, $args{cmd});
     record_info('CMD', $ssh_cmd);
     return script_output($ssh_cmd, $args{timeout}, %args);
 }
 
+=head2 scp
+
+    scp($from, $to, timeout => 90);
+
+Use scp to copy a file from or to this instance. A url starting with
+C<remote://> is replaces with the IP from this instance. E.g. a call to copy
+the file I</var/log/cloudregister> to I</tmp> looks like:
+C<<<$instance->scp('remote://var/log/cloudregister', '/tmp');>>>
+=cut
+sub scp {
+    my ($self, $from, $to, %args) = @_;
+
+    my $url = sprintf('%s@%s://', $self->username, $self->public_ip);
+    $from =~ s/^remote:\/\//$url/;
+    $to   =~ s/^remote:\/\//$url/;
+
+    my $ssh_cmd = sprintf('scp %s -i "%s" "%s" "%s"',
+        $self->ssh_opts, $self->ssh_key, $from, $to);
+
+    return script_run($ssh_cmd, $args{timeout});
+}
+
+=head2 upload_log
+
+    upload_log($filename);
+
+Upload a file from this instance to openqa using L<upload_logs()>.
+If the file doesn't exists on the instance, B<no> error is thrown.
+=cut
+sub upload_log {
+    my ($self, $remote) = @_;
```

Comment:
what about `$remote_file`, it's a bit more specific

Reply:
ok